cli/src/semgrep/commands/ci.py

```diff
@@ -82,12 +84,20 @@ def yield_valid_patterns(patterns: Iterable[str]) -> Iterable[str]:
         yield pattern
 
 
-def yield_exclude_paths(requested_patterns: Sequence[str]) -> Iterable[str]:
-    patterns = [*yield_valid_patterns(requested_patterns), *ALWAYS_EXCLUDE_PATTERNS]
-    if Path(IGNORE_FILE_NAME).is_file() and not requested_patterns:
-        patterns.extend(DEFAULT_EXCLUDE_PATTERNS)
+def get_exclude_paths(
+    requested_patterns: Optional[Mapping[out.Product, Sequence[str]]]
+) -> Mapping[out.Product, Sequence[str]]:
+    patterns = {
+        product: list(requested_patterns.get(product, [])) if requested_patterns else []
+        for product in ALL_PRODUCTS
+    }
+
+    for product in ALL_PRODUCTS:
+        patterns[product].extend(ALWAYS_EXCLUDE_PATTERNS)
+        if Path(IGNORE_FILE_NAME).is_file() and not requested_patterns:
+            patterns[product].extend(DEFAULT_EXCLUDE_PATTERNS)
```

Comment:
just for my understanding: seems like we're checking for the existence of `.semgrepignore` and if it does exist, add the default exclude patterns. Why o:?

Reply:
I just left this as-is from before, but I agree it is odd.

Reply:
You should probably put a comment here that this seems busted, but it was like this when you found it.